yt_dlp/extractor/bandcamp.py

```diff
@@ -473,12 +473,20 @@ class BandcampUserIE(InfoExtractor):
         },
     }]
 
+    def _extract_data_attr(self, webpage, video_id, attr, fatal=True):
+        return self._parse_json(self._html_search_regex(
+            fr'data-{attr}=(["\'])(\[.+?\])\1', webpage,
+            attr + ' data', group=2), video_id, fatal=fatal)
```

Comment:
I would instead use a generator method to yield all the url paths
```suggestion
    def _yield_items(self, webpage):
        yield from (
            re.findall(r'<li data-item-id=["\'][^>]+>\s*<a href=["\'](?![^"\'/]*?/merch)([^"\']+)', webpage)
            or re.finditer(r'<div[^>]+trackTitle["\'][^"\']+["\']([^"\']+)', webpage))

        yield from traverse_obj(webpage, (
            {functools.partial(get_element_html_by_id, 'music-grid')}, {extract_attributes},
            'data-client-items', {json.loads}, ..., 'page_url', {str}))
```